#------------------------------------------------------------
#revno: 8075
#author: Pete Beardmore <elbeardmorez@msn.com>
#committer: David Engster <dengste@eml.cc>
#branch nick: trunk
#timestamp: Sun 2011-08-07 13:42:47 +0100
#message:
#  (semantic-displayor-point-position): Fix case when frame is partly off
#  the screen.
=== modified file 'semantic/semantic-complete.el'
--- semantic/semantic-complete.el	2011-08-07 12:34:11 +0000
+++ semantic/semantic-complete.el	2011-08-07 12:42:47 +0000
@@ -1656,8 +1656,10 @@
   "Return the location of POINT as positioned on the selected frame.
 Return a cons cell (X . Y)"
   (let* ((frame (selected-frame))
-	 (left (frame-parameter frame 'left))
-	 (top (frame-parameter frame 'top))
+	 (left (or (car-safe (cdr-safe (frame-parameter frame 'left)))
+		   (frame-parameter frame 'left)))
+         (top (or (car-safe (cdr-safe (frame-parameter frame 'top)))
+		  (frame-parameter frame 'top)))
 	 (point-pix-pos (posn-x-y (posn-at-point)))
 	 (edges (window-inside-pixel-edges (selected-window))))
     (cons (+ (car point-pix-pos) (car edges) left)

